robottelo/host_helpers/cli_factory.py

```diff
@@ -0,0 +1,1358 @@
+"""
+This module is a more object oriented replacement for robottelo.cli.factory
+It is not meant to be used directly, but as part of a robottelo.hosts.Satellite instance
+example: my_satellite.cli_factory.make_org()
```

Comment:
Why not ?
```suggestion
example: my_satellite.cli.make_org()
```
Remove the word factory altogether ?

Reply:
cli is already an attribute that exists on the Satellite objects, it maps to the cli classes themselves.

We also can't adopt the nail gun pattern because the cli classes use class methods for their commands, so we don't actually instantiate them.

Reply:
@JacobCallahan Yeah, cli attr is in Satellite object and we should append all the make methods to that cli attr / classes. This will simplify the process of using CLI as `sat.cli.make_component` in accordance with API `sat.api.Component`.

Reply:
@jyejare it's not as simple as that. The cli attr is dynamically generated when needed, which is needed for pickle resilience.  Trying to inect these class methods into a namespace populated by classes would be messy.

Reply:
@jyejare what you're suggesting is something we could tackle by separating the hammer modeling library, but isn't reasonable with the current implementation and integration.  It is a better pattern, and one that the taretto stack suggests.